src/main/java/com/bravo/user/dao/specification/UserSpecification.java

```diff
@@ -27,5 +30,13 @@ public void doFilter(
     applyStringFilter(root.get("firstName"), filter.getFirstNames());
     applyStringFilter(root.get("lastName"), filter.getLastNames());
     applyStringFilter(root.get("middleName"), filter.getMiddleNames());
+
+    if(StringUtils.hasText(filter.getName())){
```

Comment:
This is a side-effect of using the same class to store the name.

Reply:
I would've preferred to see this validation inside the `applyStringFilterToFields` method similar to how `applyStringFilter` is implemented so if the value provided is invalid the filter is not applied at all. I think that is cleaner than validation inside the Specification itself.

Reply:
I've created a UserNameFuzzyFilter and UserNameFuzzySpecification to prevent inadvertent use of name fuzzy matching when exact name matching is desired.